cvat-ui/src/components/annotation-page/attribute-annotation-workspace/attribute-annotation-sidebar/attribute-editor.tsx

```diff
@@ -256,14 +258,56 @@ interface ListParameters {
     onChange(value: string): void;
 }
 
-function renderList(parameters: ListParameters, keyMap: ShortcutsState['keyMap']): JSX.Element | null {
+function RenderList(parameters: ListParameters): JSX.Element | null {
     const { inputType, values, onChange } = parameters;
-    if (inputType === 'checkbox') {
-        const sortedValues = ['true', 'false'];
-        if (values[0].toLowerCase() !== 'true') {
-            sortedValues.reverse();
+    const { keyMap } = useSelector((state: CombinedState) => state.shortcuts);
+    const sortedValues = ['true', 'false'];
+    const filteredValues = values.filter((value: string): boolean => value !== config.UNDEFINED_ATTRIBUTE_VALUE);
+    const prevValuesRef = useRef<string[] | null>(null);
+
+    useDynamicLabels(componentShortcuts);
+
+    useEffect(() => {
+        if (!isEqual(values, prevValuesRef.current)) {
+            const updatedComponentShortcuts = {
+                ...Object.keys(componentShortcuts).reduce((acc: any, key) => {
```

Comment:
Extra destructurization

```suggestion
            const updatedComponentShortcuts = Object.keys(componentShortcuts).reduce((acc: any, key) => {
```